gtsfm/utils/io.py

```diff
@@ -85,3 +84,80 @@ def save_json_file(
         json.dump(data, f, indent=4)
 
 
+def read_bal(file_path: str) -> GtsfmData:
+    """Read a BAL file.
+
+    Args:
+        file_name: file path of the BAL file.
+
+    Returns:
+        The data as an GtsfmData object.
+    """
+    sfm_data = gtsam.readBal(file_path)
+
+    num_images = sfm_data.number_cameras()
+
+    gtsfm_data = GtsfmData(num_images)
+    for i in range(num_images):
+        camera = sfm_data.camera(i)
+        gtsfm_data.add_camera(camera, i)
+    for j in range(sfm_data.number_tracks()):
+        gtsfm_data.add_track(sfm_data.track(j))
+
+    return gtsfm_data
+
+
+def write_cameras(gtsfm_data: GtsfmData, file_path: str) -> None:
+    """Writes the camera data file in the COLMAP format.
+
+    Args:
+        gtsfm_data: scene data to write.
+        file_path: path of the file.
+    """
+
+    # TODO: get image shape somehow
+
+    image_width = 1000  # pylint: disable=unused-variable
+    image_height = 1000  # pylint: disable=unused-variable
+
+    # TODO: handle shared intrinsics
+
+    with open(file_path, "w") as f:
+        f.write("# Number of cameras: {}\n".format(gtsfm_data.number_images()))
+
+        for i in gtsfm_data.get_valid_camera_indices():
+            camera = gtsfm_data.get_camera(i)
+            calibration = camera.calibration()
+
+            fx = calibration.fx()  # pylint: disable=unused-variable
+            u0 = calibration.px()  # pylint: disable=unused-variable
+            v0 = calibration.py()  # pylint: disable=unused-variable
+            k1 = calibration.k1()  # pylint: disable=unused-variable
+            k2 = calibration.k2()  # pylint: disable=unused-variable
+
+            f.write("{i} SIMPLE_RADIAL {image_width} {image_height} {fx} {u0} {v0} {k1} {k2}\n")
+
+
+def write_images(gtsfm_data: GtsfmData, file_path: str) -> None:
```

Comment:
same note here as above. Let's pass in `save_dir` since the file must end in `images.txt`, and otherwise we obscure this